packages/vite/src/client/client.ts

```diff
@@ -106,6 +106,7 @@ async function handleMessage(payload: HMRPayload) {
         const payloadPath = base + payload.path.slice(1)
         if (
           pagePath === payloadPath ||
+          payloadPath === '/index.html' ||
```

Comment:
Shouldn't this be `payload.path` so it also works when base isn't `'/'`? If base is `bar`, then we `payloadPath` will be `bar/index.html` and it will not hit the new condition.
```suggestion
          payload.path === '/index.html' ||
```

Reply:
nice! 🤦